presto-main/src/main/java/com/facebook/presto/sql/InterpretedFunctionInvoker.java

```diff
@@ -70,7 +71,13 @@ private Object invoke(ScalarFunctionImplementation function, ConnectorSession se
             Object argument = arguments.get(i);
             ArgumentProperty argumentProperty = function.getArgumentProperty(i);
             if (argumentProperty.getArgumentType() == VALUE_TYPE) {
-                if (function.getArgumentProperty(i).getNullConvention() == USE_NULL_FLAG) {
+                if (function.getArgumentProperty(i).getNullConvention() == RETURN_NULL_ON_NULL) {
```

Comment:
hidden bug? :)

Reply:
This won't happen before because in `ExpressionIntepreter` it's checking whether each argument's call convention is `RETURN_NULL_ON_NULL` and if that's the case & the value is null it would optimize it there so the function will never be invoked. Now that we change this call convention in metadata to be per function rather than per argument, some of those won't be optimized out and the actual execution need to take care of doing the right thing.